node/pkg/near/watcher.go

```diff
@@ -0,0 +1,395 @@
+	}
+
+	for _, o := range outcomes.Array() {
+		outcome := o.Get("outcome")
+		if !outcome.Exists() {
+			continue
+		}
+
+		executor_id := outcome.Get("executor_id")
+		if !executor_id.Exists() {
+			continue
+		}
+
+		if executor_id.String() == e.wormholeContract {
+			l := outcome.Get("logs")
+			if !l.Exists() {
+				continue
+			}
+			for _, log := range l.Array() {
+				event := log.String()
+				if !strings.HasPrefix(event, "EVENT_JSON:") {
+					continue
+				}
+				logger.Info("event", zap.String("event", event[11:]))
+
+				event_json := gjson.ParseBytes([]byte(event[11:]))
+
+				standard := event_json.Get("standard")
+				if !standard.Exists() || standard.String() != "wormhole" {
+					continue
+				}
+				event_type := event_json.Get("event")
+				if !event_type.Exists() || event_type.String() != "publish" {
+					continue
+				}
+
+				em := event_json.Get("emitter")
+				if !em.Exists() {
+					continue
+				}
+
+				emitter, err := hex.DecodeString(em.String())
+				if err != nil {
+					return err
+				}
+				//
+				if len(emitter) != 32 {
+					logger.Error("wtf")
+				}
+
+				var a vaa.Address
+				copy(a[:], emitter)
+
+				id, err := base58.Decode(hash)
+				if err != nil {
+					return err
+				}
+
+				if e.debug {
+					logger.Error("emitter: " + hex.EncodeToString(a[:]) + " id: " + hex.EncodeToString(id))
+				}
+
+				var txHash = eth_common.BytesToHash(id) // 32 bytes = d3b136a6a182a40554b2fafbc8d12a7a22737c10c81e33b33d1dcb74c532708b
+
+				v := event_json.Get("data")
+				if !v.Exists() {
+					logger.Info("data")
+					return nil
+				}
+
+				pl, err := hex.DecodeString(v.String())
+				if err != nil {
+					return err
+				}
+
+				observation := &common.MessagePublication{
+					TxHash:           txHash,
+					Timestamp:        time.Unix(int64(ts), 0),
+					Nonce:            uint32(event_json.Get("nonce").Uint()), // uint32
+					Sequence:         event_json.Get("seq").Uint(),
+					EmitterChain:     vaa.ChainIDNear,
+					EmitterAddress:   a,
+					Payload:          pl,
+					ConsistencyLevel: 0,
+				}
+
+				nearMessagesConfirmed.Inc()
+
+				logger.Info("message observed",
+					zap.Uint64("ts", ts),
+					zap.Time("timestamp", observation.Timestamp),
+					zap.Uint32("nonce", observation.Nonce),
+					zap.Uint64("sequence", observation.Sequence),
+					zap.Stringer("emitter_chain", observation.EmitterChain),
+					zap.Stringer("emitter_address", observation.EmitterAddress),
+					zap.Binary("payload", observation.Payload),
+					zap.Uint8("consistency_level", observation.ConsistencyLevel),
+				)
+
+				e.msgChan <- observation
+			}
+		}
+	}
+
+	return nil
+}
+
+func inspectBody(e *Watcher, logger *zap.Logger, block uint64, body gjson.Result) error {
+	logger.Info("inspectBody", zap.Uint64("block", block))
+
+	result := body.Get("result.chunks.#.chunk_hash")
+	if !result.Exists() {
+		return nil
+	}
+
+	v := body.Get("result.header.timestamp")
+	if !v.Exists() {
+		logger.Info("result.header.timestamp")
```

Comment:
Should this be a debug, or return an error? Same with line 244.

Reply:
Removed the debug message